### PR TITLE
Correctly mock date.constructor

### PR DESCRIPTION
**Pull request by robaw**
## What problem is this PR solving?
Currently, if you call `.constructor()` on an instance of a mocked date, you do not call the mock constructor. This change updates the constructor function on the mocked date to point to the mock date constructor. 

## Why is this necessary?
date-fns v3 [calls the constructor in this way](https://github.com/date-fns/date-fns/blob/5c1adb5369805ff552737bf8017dbe07f559b0c6/src/toDate/index.ts#L46), meaning that in order to be compatible with date-fns v3, this fix will need to be in place.

### CODE REVIEW

File: __tests__/index.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ describe('jest-date-mock', () => {
 
     // instanceof
     expect(new Date()).toBeInstanceOf(Date);
+    expect(new Date().constructor(10000)).toBeInstanceOf(Date);
```
Comment (hustcc) [inline __tests__/index.js:28]: please help to add more case.

Comment (robaw) [reply __tests__/index.js:35]: I've added some more test cases that ensure that the `.constructor()` call behaves how we'd expect.

**Comment by hustcc** [comment]
Can help to change travis to github action, like: https://github.com/hustcc/jest-random-mock/blob/main/.github/workflows/build.yml

- [ ] remove travis config
- [ ] update badge in readme.md
- [ ] add github action workflows

**Comment by robaw** [comment]
> Can help to change travis to github action, like: https://github.com/hustcc/jest-random-mock/blob/main/.github/workflows/build.yml
> 
>     * [ ]  remove travis config
> 
>     * [ ]  update badge in readme.md
> 
>     * [ ]  add github action workflows

That sounds like outside of the scope of this bugfix contribution. Maybe raise it as issues in the repo if it's work that requires contributors.

**Review by dominicfraser — APPROVED** [review]

**Comment by robaw** [comment]
@hustcc is this good to merge/release?

**Comment by hustcc** [comment]
> @hustcc is this good to merge/release?

let me run the case locally first.

**Comment by hustcc** [comment]
@robaw add github action in https://github.com/hustcc/jest-date-mock/pull/29 , you can rebase master, then ci will run in this pr.

**Comment by hustcc** [comment]
+ jest-date-mock@1.0.9

**Pull request merged**